pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -201,14 +197,39 @@ func (c *Cluster) Start() error {
 }
 
 func (c *Cluster) removeExtraMgrs(daemonIDs []string) {
+	options := metav1.ListOptions{LabelSelector: "app=" + AppName}
+	mgrDeployments, err := c.context.Clientset.AppsV1().Deployments(c.clusterInfo.Namespace).List(c.clusterInfo.Context, options)
+	if err != nil {
+		logger.Warningf("failed to check for extra mgrs. %v", err)
+		return
+	}
+	if len(mgrDeployments.Items) == len(daemonIDs) {
+		logger.Debugf("expected number %d of mgrs found", len(daemonIDs))
+		return
+	}
+
 	// In case the mgr count was reduced, delete the extra mgrs
-	for i := maxMgrCount - 1; i >= len(daemonIDs); i-- {
-		mgrName := fmt.Sprintf("%s-%s", AppName, k8sutil.IndexToName(i))
-		err := c.context.Clientset.AppsV1().Deployments(c.clusterInfo.Namespace).Delete(c.clusterInfo.Context, mgrName, metav1.DeleteOptions{})
-		if err == nil {
-			logger.Infof("removed extra mgr %q", mgrName)
-		} else if !kerrors.IsNotFound(err) {
-			logger.Warningf("failed to remove extra mgr %q. %v", mgrName, err)
+	for _, mgrDeployment := range mgrDeployments.Items {
+		id, ok := mgrDeployment.Labels[controller.DaemonIDLabel]
+		if !ok {
+			// skipping evaluation of non-mgr daemon that mistakenly matched the mgr labels
+			continue
+		}
+		found := false
+		for _, daemonID := range daemonIDs {
```

Comment:
if one of the mgr is deleted,

For ex: List of mgr{a,b,c}
b got deleted, new list {a,c}
Now a new mgr will try to get an added {a,c,d} to keep up the replica count  but as per the `daemonIDs` array is implemented it will keep the daemonIDS list as {a,b,c} only
And later replica of mgr count become = 4
New list {a,c,d,e}

So now removing the mgr will work on 
mgr list->{a,c,d,e} and hopefully daemon list should be {a,b,c,d} 
By which this algo will remove e intentionally.

I can discuss more on this

Reply:
PS: No it won't be a problem sorry for the confusion, 
It would have applied to mon algorithm